src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs

```diff
@@ -15,25 +16,52 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
 {
-    internal class FormattingContext
+    internal class FormattingContext : IDisposable
     {
-        public DocumentUri Uri { get; set; }
+        private Document _csharpWorkspaceDocument;
 
-        public DocumentSnapshot OriginalSnapshot { get; set; }
+        public DocumentUri Uri { get; private set; }
 
-        public RazorCodeDocument CodeDocument { get; set; }
+        public DocumentSnapshot OriginalSnapshot { get; private set; }
 
-        public SourceText SourceText => CodeDocument?.GetSourceText();
+        public RazorCodeDocument CodeDocument { get; private set; }
 
-        public FormattingOptions Options { get; set; }
+        public SourceText SourceText => CodeDocument.GetSourceText();
+
+        public SourceText CSharpSourceText => CodeDocument.GetCSharpSourceText();
+
+        public Document CSharpWorkspaceDocument
+        {
+            get
+            {
+                if (_csharpWorkspaceDocument == null)
+                {
+                    var adhocWorkspace = new AdhocWorkspace();
+                    var csharpOptions = adhocWorkspace.Options
+                        .WithChangedOption(CodeAnalysis.Formatting.FormattingOptions.TabSize, LanguageNames.CSharp, (int)Options.TabSize)
+                        .WithChangedOption(CodeAnalysis.Formatting.FormattingOptions.UseTabs, LanguageNames.CSharp, !Options.InsertSpaces);
+                    adhocWorkspace.TryApplyChanges(adhocWorkspace.CurrentSolution.WithOptions(csharpOptions));
+
+                    var project = adhocWorkspace.AddProject("TestProject", LanguageNames.CSharp);
```

Comment:
Is this essentially the temp project/document used during the formatting phase? Or is it related to the tests?

Reply:
This is just a temporary project used for formatting purposes. I tried to avoid it but looks like it needs a project associated with it to work properly.